src/rbindlist.c

```diff
@@ -225,19 +225,21 @@ SEXP rbindlist(SEXP l, SEXP usenamesArg, SEXP fillArg, SEXP idcolArg)
         if (isNull(getAttrib(thisCol,R_LevelsSymbol))) error("Column %d of item %d has type 'factor' but has no levels; i.e. malformed.", w+1, i+1);
         factor=true;   // TODO isOrdered(thiscol) ? 2 : 1;
       } //else if (!isString(thisCol) && length(thisCol)) anyNotStringOrFactor=true;
-      SEXP thisClass = getAttrib(thisCol, R_ClassSymbol);
-      if (INHERITS(thisClass, char_integer64)) int64=true;
+      if (INHERITS(thisCol, char_integer64)) {
+        if (firsti>=0 && !length(getAttrib(firstCol, R_ClassSymbol))) { firsti=i; firstw=w; firstCol=thisCol; } // so the integer64 attribute gets copied to target below
+        int64=true;
+      }
       if (firsti==-1) { firsti=i; firstw=w; firstCol=thisCol; }
-      else if (!factor && !R_compute_identical(thisClass, getAttrib(firstCol,R_ClassSymbol), 0)) {
+      else if (!factor && !int64 && !R_compute_identical(getAttrib(thisCol, R_ClassSymbol), getAttrib(firstCol,R_ClassSymbol), 0)) {
         error("Class attribute on column %d of item %d does not match with column %d of item %d.", w+1, i+1, firstw+1, firsti+1);
       }
     }
     // in future warn ... if (factor && anyNotStringOrFactor) warning("Column %d contains a factor but not all items for the column are character or factor", idcol+j+1);
     if (!foundName) { char buff[12]; sprintf(buff,"V%d",j+1), SET_STRING_ELT(ansNames, idcol+j, mkChar(buff)); }
     if (factor) maxType=INTSXP;  // any items are factors then a factor is created (could be an option)
+    if (int64 && maxType!=REALSXP) error("Internal error: column %d of result is determined to be integer64 but maxType==%d != REALSXP", j+1, maxType); // nocov
```

Comment:
Missing `#` near nocov

Reply:
Good sport - thanks. Looks like C coverage doesn't distinguish parts of lines, unlike R coverage.